Integrations/integration-CounterTack.yml

```diff
@@ -0,0 +1,1693 @@
+                msg = msg + "\n"+json.dumps(res.get('details')[0])
+        except Exception as e:
+            return response.text
+        return msg
+
+
+    """
+
+    ENDPOINTS
+
+    """
+
+
+    def get_endpoints_request():
+        """
+        This request returns a collection of endpoints.
+        """
+        suffix_url = 'endpoints'
+        response = http_request('GET', suffix_url)
+        return response.json()
+
+
+    def get_endpoints():
+
+        """
+        Returns the information on existing endpoints
+        """
+        data = []
+        headers = ['OS', 'Name', 'Threat', 'Status', 'ID', 'IP']
+        endpoints = get_endpoints_request()
+        for endpoint in endpoints:
+            data.append({
+                'ID': endpoint.get('id'),
+                'Name': endpoint.get('name'),
+                'OS': endpoint.get('product_name'),
+                'IP': endpoint.get('ips'),
+                'Status': endpoint.get('status'),
+                'Threat': endpoint.get('threat')
+            })
+            endpoint_standards = {
+                'ID': endpoint.get('id'),
+                'IPAddress': endpoint.get('ips'),
+                'Domain': endpoint.get('domain'),
+                'MACAddress': endpoint.get('mac'),
+                'OS': endpoint.get('product_name'),
+                'OSVersion': endpoint.get('driver_version'),
+                'Model': endpoint.get('current_profile'),
+                'Memory': endpoint.get('memory'),
+                'Processors': endpoint.get('num_cpus')
+            }
+
+        context = {
+            'CounterTack.Endpoint(val.id && val.id === obj.id)': endpoints
+        }
+
+        context = {
+            'Endpoint(val.id && val.id === obj.id)': endpoint_standards
+        }
+
+        entry = {
+            'Type': entryTypes['note'],
+            'Contents': endpoints,
+            'ContentsFormat': formats['json'],
+            'ReadableContentsFormat': formats['markdown'],
+            'HumanReadable': tableToMarkdown(
+                'CounterTack Endpoints', data, headers, removeNull=True),
+            'EntryContext': context
+        }
+        demisto.results(entry)
+
+
+    def get_endpoint_request(endpoint_id):
+
+        """
+        Request for a specific endpoint
+        """
+        suffix_url = 'endpoints/' + endpoint_id
+
+        response = http_request('GET', suffix_url)
+        return response.json()
+
+
+    def get_endpoint():
+        """
+        Get the information for the requested endpoint
+
+        demisto parameter: (string) endpoint_id
+            The unique ID of the endpoint
+
+        returns:
+            The information about the specified endpoint
+        """
+        headers = ['OS', 'Domain', 'IP', 'Threat', 'MaxImpact', 'TenantID', 'IsQuarantined', 'Profile', 'Tags', 'Cluster_Hosts', 'Status']
+        endpoint_id = demisto.args().get('endpoint_id')
+        response = get_endpoint_request(endpoint_id)
```

Comment:
should check if resposne or if response is instance of dict or else the .get() will throw exception